crates/frontend/src/logic.rs

```diff
@@ -0,0 +1,278 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Map, Value};
+
+use crate::schema::{HtmlDisplay, SchemaType};
+use derive_more::{Deref, DerefMut};
+use superposition_types::Context;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum Operand {
+    Value(serde_json::Value),
+    Dimension(serde_json::Value),
+}
+
+impl Operand {
+    pub fn from_operand_json(value: Value) -> Self {
+        match value {
+            Value::Object(ref o) if o.contains_key("var") => Operand::Dimension(value),
+            v => Operand::Value(v),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deref, DerefMut, Serialize, Deserialize)]
+pub struct Operands(pub Vec<Operand>);
+
+impl FromIterator<Value> for Operands {
+    fn from_iter<T: IntoIterator<Item = Value>>(iter: T) -> Self {
+        Operands(
+            iter.into_iter()
+                .map(Operand::from_operand_json)
+                .collect::<Vec<Operand>>(),
```

Comment:
Can we make a type for Vec<Operand> as Operands as we made for Conditions ?

Reply:
Didn't really understand the point.